src/main/java/edu/ie3/datamodel/io/source/couchbase/CouchbaseWeatherSource.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.source.couchbase;
+
+import com.couchbase.client.core.error.DecodingFailureException;
+import com.couchbase.client.core.error.DocumentNotFoundException;
+import com.couchbase.client.java.json.JsonObject;
+import com.couchbase.client.java.kv.GetResult;
+import com.couchbase.client.java.query.QueryResult;
+import edu.ie3.datamodel.io.connectors.CouchbaseConnector;
+import edu.ie3.datamodel.io.factory.timeseries.TimeBasedWeatherValueData;
+import edu.ie3.datamodel.io.factory.timeseries.TimeBasedWeatherValueFactory;
+import edu.ie3.datamodel.io.source.IdCoordinateSource;
+import edu.ie3.datamodel.io.source.WeatherSource;
+import edu.ie3.datamodel.models.timeseries.individual.IndividualTimeSeries;
+import edu.ie3.datamodel.models.timeseries.individual.TimeBasedValue;
+import edu.ie3.datamodel.models.value.WeatherValue;
+import edu.ie3.util.interval.ClosedInterval;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.locationtech.jts.geom.Point;
+
+/** Couchbase Source for weather data */
+public class CouchbaseWeatherSource implements WeatherSource {
+  private static final Logger logger = LogManager.getLogger(CouchbaseWeatherSource.class);
+  private static final String COORDINATE_ID_COLUMN_NAME = "coordinate";
+  private static final String TIMESTAMP_PATTERN = "yyyy-MM-dd'T'HH:mm:ssxxx";
```

Comment:
Same - pls don't hard code important parameters but le the user provide them in a parametrized constructor. 
As the timestamp pattern is indeed a "classic" one, here a second constructor with this default pattern IMHO makes sense tough.